src/components/editor/index.js

```diff
@@ -0,0 +1,261 @@
+import { PropTypes } from 'react';
+import reactStamp from 'react-stamp';
+import { fromJS, List, Repeat } from 'immutable';
+import Editor from 'draft-js-plugins-editor';
+import {
+  EditorState,
+  ContentState,
+  ContentBlock,
+  convertFromHTML,
+  convertToRaw,
+  convertFromRaw,
+  RichUtils,
+  Modifier,
+  BlockMapBuilder,
+  genKey,
+  CharacterMetadata,
+} from 'draft-js';
+
+function insertFragment(editorState, fragment, entityMap) {
+  var newContent = Modifier.replaceWithFragment(
+    editorState.getCurrentContent(),
+    editorState.getSelection(),
+    fragment
+  );
+
+  return EditorState.push(
+    editorState,
+    newContent,
+    'insert-fragment'
+  );
+}
+
+const EditorComponent = (React, ...behaviours) => reactStamp(React).compose({
+  propTypes: {
+    content: PropTypes.shape({
+      entityMap: PropTypes.object.isRequired,
+      blocks: PropTypes.array.isRequired,
+    }).isRequired,
+
+    onEditStart: PropTypes.func,
+    onEditEnd: PropTypes.func,
+    onChange: PropTypes.func,
+
+    readOnly: PropTypes.bool,
+    plugins: PropTypes.array,
+  },
+
+  init() {
+    const content = convertFromRaw(this.props.content);
+    this.state = {
+      editor: EditorState.createWithContent(content),
+      readOnly: true,
+    };
+
+    this.onClick = this.onClick.bind(this);
+    this.onChange = this.onChange.bind(this);
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+
+    this.handleKeyCommand = this.handleKeyCommand.bind(this);
+    this.handleReturn = this.handleReturn.bind(this);
+    this.handlePastedText = this.handlePastedText.bind(this);
+
+    this.save = this.save.bind(this);
+  },
+
+  componentWillUnmount() {
+    this.deleteTimeout();
+    this.save();
+  },
+
+  onClick() {
+    this.setState({ readOnly: false }, () => {
+      this.refs.upstream && this.refs.upstream.focus();
```

Comment:
I would argue unit testing is barely making sense in this particular case. The purpose of unit testing is test behavior of a particular unit, but the scope of unit testing is concerned only with public API. `_focus` is anything but public API, as its implementation and even the underscore-prefixed name suggest. The only public API of React components is its rendered data.

It does pass the feel-good check of having everything tested, but I want to ask, what's the value of this test? Like, really.

The primary value of testing lies in automation of manual testing process, which is both tedious and time consuming. The most natural thing to test, so far as many things concerning the UI, is what component renders and how it reacts to the user interactions — by literally encoding in code how the user would interact and how the rendered content would change as the result. Which is literally what integration tests do.

Unit tests, on another hand, are most valuable when you have a clear input -> output, like making sure `add(2, 2)` produces `4`, and so on.

It's a bit of a mystery to me as to why you insist on having all of that unit-tested, instead of integration-tested which clearly seems to make more sense. 

I can make all the proposed changes if only to move this PR forward as it's already taking more time than I imagined it would